src/http/matcher/mod.rs

```diff
@@ -65,9 +68,30 @@ pub enum HttpMatcherKind {
     ///
     /// [`SocketAddr`]: std::net::SocketAddr
     Socket(SocketMatcher),
+    /// A custom matcher that implements [`crate::service::Matcher`].
+    // Custom(Box<dyn crate::service::Matcher<State, Request<Body>>>), TODO: Does not satisfy Clone, unless inner field satisfies Clone.
+    Custom(Arc<dyn crate::service::Matcher<State, Request<Body>>>),
```

Comment:
As discussed you can use `Box<>` not `Arc<>`.

Reply:
As per more recent [comment](https://github.com/dmackdev/rama/pull/1#discussion_r1545646890), using `Arc`.